app/models/document.rb

```diff
@@ -52,4 +50,10 @@ def do_not_sanitize_these_fields
   end
 
   attr_reader :document
+
+  def keys_to_hash(keys)
+    hash = {}
+    keys.each { |key| hash[key] = get_value(key) }
+    hash
```

Comment:
Could this be simplified to?

```suggestion
    keys.to_h { |key| [key, get_value(key)] }
```